packit/sync.py

```diff
@@ -1,127 +1,81 @@
-# MIT License
-#
-# Copyright (c) 2019 Red Hat, Inc.
-
-# Permission is hereby granted, free of charge, to any person obtaining a copy
-# of this software and associated documentation files (the "Software"), to deal
-# in the Software without restriction, including without limitation the rights
-# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-# copies of the Software, and to permit persons to whom the Software is
-# furnished to do so, subject to the following conditions:
-#
-# The above copyright notice and this permission notice shall be included in all
-# copies or substantial portions of the Software.
-#
-# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-# SOFTWARE.
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
 
 import logging
-import shutil
 from pathlib import Path
-from typing import List
-from typing import NamedTuple, Union
+from typing import List, Optional, Union, Sequence
 
 from packit.exceptions import PackitException
+from packit.utils import run_command
 
 logger = logging.getLogger(__name__)
 
 
-class SyncFilesItem(NamedTuple):
-    src: Union[str, List[str]]
-    dest: str
+def check_subpath(subpath: Path, path: Path) -> Path:
+    """Check if 'subpath' is a subpath of 'path'
+    Args:
+        subpath: Subpath to be checked.
+        path: Path agains which subpath is checked.
+    Returns:
+        'subpath', in case it is a subpath of 'path'.
+    Raises:
+        PackitException, if 'subpath' is not a subpath of 'path'.
+    """
+    subpath_resolved = subpath.resolve()
+    path_resolved = path.resolve()
+    if not str(subpath_resolved).startswith(str(path_resolved)):
+        raise PackitException(
+            f"Sync files: Illegal path! {subpath} is not a subpath of {path}."
```

Comment:
Looking at [Path.relative_to()](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to) it raises basically the same exception message. Would that make the function a bit more simple?

Reply:
Right, I totally got confused about this. Thanks for pointing it out! I would go with using `Path.is_relative_to()` and raise a `PackitException` if it's false, with this error message, just to be able to get rid of the `OR one path is relative and the other is absolute`-part of the `ValueError` raised by `pathlib`. I think this might be confusing to users, and this function is taking care of that part already.